packages/react/use-loading/src/useLoading.ts

```diff
@@ -1,16 +1,37 @@
 /** @tossdocs-ignore */
-import { useCallback, useMemo, useState } from 'react';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 
 export function useLoading(): [boolean, <T>(promise: Promise<T>) => Promise<T>] {
   const [loading, setLoading] = useState(false);
-  const startTransition = useCallback(async <T>(promise: Promise<T>) => {
-    try {
-      setLoading(true);
-      const data = await promise;
-      return data;
-    } finally {
-      setLoading(false);
-    }
-  }, []);
+  const ref = useIsMountedRef();
+
+  const startTransition = useCallback(
+    async <T>(promise: Promise<T>) => {
+      try {
+        setLoading(true);
+        const data = await promise;
+        return data;
+      } finally {
+        if (ref.isMounted) {
+          setLoading(false);
+        }
+      }
+    },
+    [ref.isMounted]
+  );
+
   return useMemo(() => [loading, startTransition], [loading, startTransition]);
 }
+
+function useIsMountedRef() {
+  const ref = useRef({ isMounted: true }).current;
+
+  useEffect(() => {
+    ref.isMounted = true;
+    return () => {
+      ref.isMounted = false;
+    };
+  }, [ref]);
+
+  return ref;
+}
```

Comment:
Why don't you add this hook to `@toss/react` ?!

It is pretty cool hook to user who manage state regardless rendering lifecycle

Reply:
thx I think it's a `hook` that can be used to prevent unwanted state updates in many situations